.github/actions/setup-rust/setup.sh

```
@@ -0,0 +1,31 @@
+# Source cargo env for this script
+. "$HOME/.cargo/env"
+
+# Install components if specified
+if [ -n "$COMPONENTS" ]; then
+	# Install each component individually
+	echo "$COMPONENTS" | tr ',' '\n' | while read -r component; do
+		component=$(echo "$component" | tr -d ' ')
+		[ -n "$component" ] && rustup component add "$component"
+	done
+fi
+fi
```

Shellcheck (0.11.0)

[error] Parsing stopped here. Is this keyword correctly matched up?

(SC1089)